frigate/http.py

```diff
@@ -1008,6 +1009,48 @@ def config_save():
         return "Config successfully saved.", 200
 
 
+@bp.route("/config/set", methods=["PUT"])
+def config_set():
+    config_file = os.environ.get("CONFIG_FILE", f"{CONFIG_DIR}/config.yml")
+
+    # Check if we can use .yaml instead of .yml
+    config_file_yaml = config_file.replace(".yml", ".yaml")
+
+    if os.path.isfile(config_file_yaml):
+        config_file = config_file_yaml
+
+    with open(config_file, "r") as f:
+        old_raw_config = f.read()
+        f.close()
+
+    try:
+        update_yaml_from_url(config_file, request.url)
```

Comment:
The config file should not be saved to before the config is verified. In the event an error occurred the user would be unable to start frigate without editing the config in some other way.

Reply:
Look to lines 1032-1036: if the configuration is wrong, it is rolled back